ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/ImportBlocksStepTest.java

```diff
@@ -113,9 +115,41 @@ public void shouldThrowExceptionWhenValidationFails() {
             blockWithReceipts.getReceipts(),
             FULL,
             LIGHT,
-            BodyValidationMode.LIGHT))
+            BodyValidationMode.LIGHT,
+            true))
         .thenReturn(new BlockImportResult(false));
     assertThatThrownBy(() -> importBlocksStep.accept(singletonList(blockWithReceipts)))
         .isInstanceOf(InvalidBlockException.class);
   }
+
+  @Test
+  public void shouldImportBlockWithoutTxIndexingWhenNotEnabled() {
+    ImportBlocksStep importBlocksStep =
+        new ImportBlocksStep(
+            protocolSchedule,
+            protocolContext,
+            validationPolicy,
+            ommerValidationPolicy,
+            null,
+            pivotHeader,
+            false);
+
+    final Block block = gen.block();
+    final BlockWithReceipts blockWithReceipts = new BlockWithReceipts(block, gen.receipts(block));
+
+    when(blockImporter.importBlockForSyncing(
+            protocolContext,
+            blockWithReceipts.getBlock(),
+            blockWithReceipts.getReceipts(),
+            FULL,
+            LIGHT,
+            BodyValidationMode.LIGHT,
+            false))
+        .thenReturn(new BlockImportResult(true));
+
+    importBlocksStep.accept(List.of(blockWithReceipts));
+
+    verify(protocolSchedule).getByBlockHeader(blockWithReceipts.getHeader());
+    verify(validationPolicy, times(1)).getValidationModeForNextBlock();
```

Comment:
Do you want to add a check that the transaction has not been indexed?

Reply:
Yes that's a good test to add

Reply:
Actually that test already exists as `shouldImportBlocks`.

Reply:
I've added a verify to make it a better test though as it just had a stub for that condition